elements/pf-back-to-top/pf-back-to-top.ts

```diff
@@ -0,0 +1,154 @@
+ * @cssprop {<length>} --pf-c-button--PaddingTop {@default `0.25rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingRight {@default `0.5rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingBottom {@default `0.25rem`}
+ * @cssprop {<length>} --pf-c-button--PaddingLeft {@default `0.5rem`}
+ * @cssprop {<color>} --pf-c-button--m-primary--Color {@default `#fff`}
+ * @cssprop {<color>} --pf-c-button--m-primary--BackgroundColor {@default `#06c`}
+ * @cssprop {<length>} --pf-c-button__icon--m-end--MarginLeft {@default `0.25rem`}
+ *
+ */
+@customElement('pf-back-to-top')
+export class PfBackToTop extends LitElement {
+  static readonly styles = [styles];
+
+  #scrollSpy = false;
+
+  #visible = false;
+
+  #scrollElement?: HTMLElement | Window;
+
+  #logger = new Logger(this);
+
+  /** Shorthand for the `icon` slot, the value is icon name */
+  @property({ reflect: true }) icon?: string;
+
+  /** Icon set for the `icon` property */
+  @property({ attribute: 'icon-set' }) iconSet?: string;
+
+  /** Flag to always show back to top button, defaults to false. */
+  @property({ type: Boolean, reflect: true, attribute: 'always-visible' }) alwaysVisible = false;
+
+  /** Element selector to spy on for scrolling. Not passing a selector defaults to spying on window scroll events */
+  @property({ reflect: true, attribute: 'scrollable-selector' }) scrollableSelector?: string;
+
+  /** Distance from the top of the scrollable element to trigger the visibility of the back to top button */
+  @property({ type: Number, attribute: 'scroll-distance' }) scrollDistance = 400;
+
+  /** Page fragment link to target element, must include hash ex: #top */
+  @property({ reflect: true }) href?: string;
+
+  get #rootNode(): Document | ShadowRoot {
+    const root = this.getRootNode();
+    if (root instanceof Document || root instanceof ShadowRoot) {
+      return root;
+    } else {
+      return document;
+    }
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.#scrollSpy = !!this.scrollableSelector;
+
+    if (this.scrollableSelector?.trim() === '') {
+      this.#logger.error(`scrollable-selector attribute cannot be empty`);
+      return;
+    }
+
+    if (this.#scrollSpy && !!this.scrollableSelector) {
+      const scrollableElement = this.#rootNode.querySelector(this.scrollableSelector) as HTMLElement;
+      if (!scrollableElement) {
+        this.#logger.error(`Unable to find element with selector ${this.scrollableSelector}`);
+        return;
+      }
+      this.#scrollElement = scrollableElement;
+    } else {
+      this.#scrollElement = window;
+    }
+
+    this.#scrollElement.addEventListener('scroll', this.#toggleVisibility, { passive: true });
+    this.#toggleVisibility();
+  }
+
+  override disconnectedCallback(): void {
+    super.disconnectedCallback?.();
+    this.#scrollElement?.removeEventListener('scroll', this.#toggleVisibility);
```

Comment:
make sure to also do this at the top of the reaction callback from the previous comment